### PR TITLE
Worker v1

### PR DESCRIPTION
**Pull request by nord94**
Unfortunatelly, this piece of shit doesnt work. Look pls

### CODE REVIEW

File: src/javase02/task01/Main.java [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package javase02.task01;
```
Comment (alexeykotelevskiy) [inline src/javase02/task01/Main.java:1]: General comment: append to .gitignore out/ directory

File: src/javase02/task01/Main.java [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+
```
Comment (alexeykotelevskiy) [inline src/javase02/task01/Main.java:2]: General comment: use ctrl+alt+L for each file before commit.

File: src/javase02/task02/Chancellery.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
+package javase02.task02;
+
+public class Chancellery {
+    int price;
+    int number;
+}
+
+class Pen extends Chancellery{
```
Comment (alexeykotelevskiy) [inline src/javase02/task02/Chancellery.java]: Empty class seems strange. Maybe better create class Pen and Enum with colors?

File: src/javase02/task02/Worker.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
+package javase02.task02;
+
+import java.util.Scanner;
+
+public class Worker {
+    String name;
+    int numberOfChancellery;
+    Chancellery[] chancelleries;
+    //Chancellery[] chancelleries = new Chancellery[numberOfChancellery];
```
Comment (alexeykotelevskiy) [inline src/javase02/task02/Worker.java]: remove commented lines

File: src/javase02/task02/Worker.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    static Scanner input = new Scanner(System.in);
+    private int calculatePriceOfChancellery(Worker obj){
+        int summ=0;
+        for (int i =0;i<numberOfChancellery;i++) summ+=obj.chancelleries[i].price;
+        return summ;
+    }
+
+    Worker(String name1, int number){
```
Comment (alexeykotelevskiy) [inline src/javase02/task02/Worker.java]: name1 is bad name

File: src/javase02/task02/Worker.java [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+        name=name1;
+        numberOfChancellery=number;
+        chancelleries = new Chancellery[number];
+    }
+
+    void collectChancellery(){
+        Chancellery[] chancelleries1 = new Chancellery[5];
```
Comment (alexeykotelevskiy) [inline src/javase02/task02/Worker.java]: chancelleries1 is bad name

File: src/javase02/task02/Worker.java [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+        chancelleries1[2]=new GreenPen();
+        chancelleries1[3]=new BlackPencil();
+        chancelleries1[4]=new WhitePencil();
+
+        int overStep=0;
+        for (int i=0;i<numberOfChancellery;i++){
+            if (overStep==0){
+                System.out.println("Enter: 1 - BluePen, 2 - GreenPen, 3 - RedPen" +
+                        "4 - BlackPencil, 5 - WhitePencil");
+                int itemIndex = input.nextInt();
```
Comment (alexeykotelevskiy) [inline src/javase02/task02/Worker.java]: need validation

File: src/javase02/task02/Worker.java [hunk 8]
```diff
@@ -0,0 +1,57 @@
+package javase02.task02;
+
+import java.util.Scanner;
+
+public class Worker {
+    String name;
+    int numberOfChancellery;
```
Comment (alexeykotelevskiy) [inline src/javase02/task02/Worker.java]: seems like numberOfChancellery no needed. chancelleries.length is enough

File: src/javase02/task02/Chancellery.java (hunk 3 above)
Comment (alexeykotelevskiy) [reply src/javase02/task02/Chancellery.java]: or extend RedPen, GreenPen etc. directly from  Chancellery

File: src/javase02/task02/Worker.java [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+    void collectChancellery(){
+        Chancellery[] chancelleries1 = new Chancellery[5];
+        chancelleries1[0]=new BluePen();
+        chancelleries1[1]=new RedPen();
+        chancelleries1[2]=new GreenPen();
+        chancelleries1[3]=new BlackPencil();
+        chancelleries1[4]=new WhitePencil();
+
+        int overStep=0;
+        for (int i=0;i<numberOfChancellery;i++){
```
Comment (alexeykotelevskiy) [inline src/javase02/task02/Worker.java]: Im not understood what are doing in this cycle. Please add comments

File: src/javase02/task02/Worker.java [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+
+import java.util.Scanner;
+
+public class Worker {
+    String name;
+    int numberOfChancellery;
+    Chancellery[] chancelleries;
+    //Chancellery[] chancelleries = new Chancellery[numberOfChancellery];
+    static Scanner input = new Scanner(System.in);
+    private int calculatePriceOfChancellery(Worker obj){
```
Comment (alexeykotelevskiy) [inline src/javase02/task02/Worker.java]: good point for creating unit test

File: src/javase02/task02/Chancellery.java [hunk 11]
```diff
@@ -0,0 +1,44 @@
+package javase02.task02;
+
+public class Chancellery {
```
Comment (alexeykotelevskiy) [inline src/javase02/task02/Chancellery.java:3]: if you assume that Chancellery is abstract class, append 'abstract' specifier

File: src/javase02/task01/Main.java (hunk 1 above)
Comment (nord94) [reply src/javase02/task01/Main.java:1]: done 👌

File: src/javase02/task01/Main.java (hunk 2 above)
Comment (nord94) [reply src/javase02/task01/Main.java:2]: got it 😎

File: src/javase02/task02/Chancellery.java (hunk 3 above)
Comment (nord94) [reply src/javase02/task02/Chancellery.java]: I made them abstract. Then i added isUsesInk and isUsesGraphite for both of them.

File: src/javase02/task02/Worker.java (hunk 4 above)
Comment (nord94) [reply src/javase02/task02/Worker.java]: yup

File: src/javase02/task02/Worker.java (hunk 5 above)
Comment (nord94) [reply src/javase02/task02/Worker.java]: I know. I can use _this_-keyword in this constructor. Is it a good idea?

File: src/javase02/task02/Worker.java (hunk 6 above)
Comment (nord94) [reply src/javase02/task02/Worker.java]: Same thing as with name1

File: src/javase02/task02/Worker.java (hunk 8 above)
Comment (nord94) [reply src/javase02/task02/Worker.java]: yup

File: src/javase02/task02/Worker.java (hunk 10 above)
Comment (nord94) [reply src/javase02/task02/Worker.java]: For now i dont know how to make them.

File: src/javase02/task02/Chancellery.java (hunk 11 above)
Comment (nord94) [reply src/javase02/task02/Chancellery.java:3]: I wont do this cause it wil forbid making instances of it

File: src/javase02/task02/Worker.java (hunk 7 above)
Comment (nord94) [reply src/javase02/task02/Worker.java]: Debug? Is that what u mean?

File: src/javase02/task02/Worker.java (hunk 5 above)
Comment (alexeykotelevskiy) [reply src/javase02/task02/Worker.java]: yes

File: src/javase02/task02/Worker.java (hunk 10 above)
Comment (alexeykotelevskiy) [reply src/javase02/task02/Worker.java]: https://junit.org/junit5/docs/current/user-guide/

File: src/javase02/task02/Worker.java (hunk 7 above)
Comment (alexeykotelevskiy) [reply src/javase02/task02/Worker.java]: i mean what happened if user write 228 or 1488? Will be IndexOfBound exception. 
You should check input data

**Pull request merged**